node/v2/error_response.js

```diff
@@ -65,8 +65,31 @@ CodeNames[Codes.NetworkError] = 'network error';
 CodeNames[Codes.ProtocolError] = 'protocol error';
 CodeNames[Codes.Unhealthy] = 'unhealthy';
 
+function CodeError(type, message, isErrorFrame, codeName, errorCode, originalId) {
+    var self = this;
+    self.type = type;
+    self.message = message;
+    self.isErrorFrame = isErrorFrame;
+    self.codeName = codeName;
+    self.errorCode = errorCode;
+    self.originalId = originalId;
+}
+
+function partiallyApplyCodeError(options1) {
+    return function makeError(options2) {
```

Comment:
I'd be much happier if we could just have a raft of mundane prototype'd objects for more clarity, and also not letting instantiators specify fixed fields like type, message, etc... how about:
```
function FooError(originalId) {
  // assign constant type, message, isErrorFrame, etc up to:
  this.origingalId = originalId;
}
```

it should actually less total loc than this current pr diff just to have N-many constructors declared explicitly.